### PR TITLE
feat: allow running review app to validate pr from external contribution

### PR DESCRIPTION
**Pull request by FabienArcellier**
The integration of an app review mechanism deploys a PR on demand to test the code online or demonstrate it without requiring cloning the branch locally.

Peek 2024-02-19 18-05

I used the reviewapp mechanism from [scalingo](https://scalingo.com/), a French PaaS platform. Their implementation is close to Heroku. After a first reading, project administrators will be able to trigger an environment to try and experiment on the code of the PR.

Peek 2024-02-19 17-24

The validation environment can be modified by any visitor. If this mechanism poses a problem, we will add authentication to limit access and only offer read access for visitors.

### CODE REVIEW

File: apps/reviewapp.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import os
```
Comment (FabienArcellier) [inline apps/reviewapp.py:1]: Add a header to this module to explain why we have created the reviewapp.

File: .buildpacks [hunk 2]
```diff
@@ -0,0 +1,2 @@
+https://github.com/FabienArcellier/nodejs-buildpack#streamsync-review
```
Comment (FabienArcellier) [inline .buildpacks:1]: I don't know where to document the role of the Procfile and .buildpacks. ``.buildpacks`` does not support code comment.

**Comment by ramedina86** [comment]
> The validation environment can be modified by any visitor. If this mechanism poses a problem, we will add authentication to limit access and only offer read access for visitors.

Is it too hard to implement auth with something like Google Auth and whitelist a few people?

Running `streamsync run` for everyone and `streamsync edit` for whitelisted would be ideal, but I'm sure harder to implement.

We'll be deploying many features that'll require us to handle API keys and maybe other sorts of confidential information. Being overly cautious but I'm scared someone could install a keylogger / etc, dump memory, or do something nasty.

File: pyproject.toml [hunk 3]
```diff
@@ -34,6 +34,7 @@ watchdog = ">= 3.0.0, < 4"
 pandas = {version = ">= 2.2.0, < 3", optional = true}
 pyarrow = {version = ">= 15.0.0, < 16.0.0",optional = true}
 plotly = {version = ">= 5.18.0, < 6", optional = true}
+scikit-learn = {version = "^1.4.1.post1", optional = true}
```
Comment (ramedina86) [inline pyproject.toml:37]: Why is this here? Do we require it now for any reason?

Comment (FabienArcellier) [reply pyproject.toml:37]: We are using it in the getting started application.

**Comment by ramedina86** [comment]
Hey @FabienArcellier what's the status of this? Are we going for the custom buildpack? If it's too hard we can even drop the quickstart app, I wouldn't spend too much time / add too much complexity to support this

**Comment by FabienArcellier** [comment]
I have finished the work on the buildpack. This branch is not required anymore. I close it.
